bundles/org.jkiss.utils/src/org/jkiss/utils/IOUtils.java

```diff
@@ -441,4 +441,18 @@ public static boolean isLocalPath(Path filePath) {
         return isLocalURI(filePath.toUri());
     }
 
+    /**
+     * Returns the <em>grandparent path</em>, or {@code null} if this path does not have a grandparent.
+     *
+     * @param path a path
+     * @return a path representing the path's grandparent
+     */
+    @Nullable
+    public static Path getGrandparent(@NotNull Path path) {
```

Comment:
This method seems very specific and only used in a single place.

Reply:
I'll reuse it a bit later (PR is not ready yet)